src/main/java/uk/gov/hmcts/befta/player/DefaultBackEndFunctionalTestScenarioPlayer.java

```diff
@@ -621,12 +632,54 @@ private void authenticateUser(final BackEndFunctionalTestScenarioContext scenari
             UserData user) {
         String logPrefix = scenarioContext.getTestData().get_guid_() + ": " + prefix + " [" + user.getUsername() + "] ";
         String preferredTokenProviderClientId = scenarioContext.getTestData().getUserTokenClientId();
+
+        AuthenticationRetryConfiguration authenticationRetryConfiguration =
+                BeftaMain.getConfig().getAuthenticationRetryConfiguration();
+
         scenario.log("Authentication attempt from: " + preferredTokenProviderClientId + ".");
-        try {
+        if (authenticationRetryConfiguration.isRetryDisabled()) {
+            try {
+                BeftaMain.getAdapter().authenticate(user, preferredTokenProviderClientId);
+                BeftaUtils.defaultLog(logPrefix + "authenticated.");
+            } catch (Exception ex) {
+                throw new FunctionalTestException(logPrefix + "could not authenticate.", ex);
+            }
+        } else {
+            BeftaUtils.defaultLog("Authentication retry enabled");
+            authenticateUserWithRetry(authenticationRetryConfiguration, user, logPrefix, preferredTokenProviderClientId);
+        }
+    }
+
+    private void authenticateUserWithRetry(AuthenticationRetryConfiguration config,
+                                           UserData user,
+                                           String logPrefix,
+                                           String preferredTokenProviderClientId) {
+        AtomicInteger counter = new AtomicInteger(0);
+        Callable<Boolean> callable = () -> {
+            logger.info("Number of retries while attempting to authenticate user {} time(s).",counter.incrementAndGet());
```

Comment:
Better like logger.info("Attempting to authenticate {}. Retry no = {}.", user.getUsername(), counter.incrementAndGet())